addons/fix-uploaded-svgs/userscript.js

```diff
@@ -1,4 +1,101 @@
 export default async function ({ addon, console }) {
+  // Fixes "x", "y", and "dominant-baseline" being ignored by Scratch, "span" causing offset
+  function fixEditorSVG(element) {
+    const svg = element.cloneNode(true);
+
+    // This fixes the previous issue of SVGs imported from Affinity Designer
+    if (svg.height.baseVal.valueAsString === "100%" && svg.width.baseVal.valueAsString === "100%") {
+      svg.removeAttribute("height");
+      svg.removeAttribute("width");
+    }
+
+    // This iframe is needed to correct "dominant-baseline"
+    const iframe = document.createElement("iframe");
+    iframe.setAttribute("src", "about:blank");
+    document.body.append(iframe);
+    iframe.contentDocument.body.appendChild(svg);
+
+    let transform = "",
+      translate = "",
+      translateIndex = 0,
+      x = "",
+      y = "",
+      difference = 0,
+      baselineAdjust = "",
+      innerTextElement = "",
+      innerTextSpan = "";
```

Comment:
I meant defining most of these during their first assignment, but for `transform` and `translate` it makes sense to define them without a value here or at the top of the loop.